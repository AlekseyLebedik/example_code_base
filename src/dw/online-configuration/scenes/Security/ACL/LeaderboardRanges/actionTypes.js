export const LEADERBOARD_RANGES_FETCH = 'security/LEADERBOARD_RANGES_FETCH';
export const LEADERBOARD_RANGES_FETCH_SUCCESS =
  'security/LEADERBOARD_RANGES_FETCH_SUCCESS';

export const LEADERBOARD_RANGE_ADD = 'security/LEADERBOARD_RANGE_ADD';
export const LEADERBOARD_RANGE_ADD_SUCCESS =
  'security/LEADERBOARD_RANGE_ADD_SUCCESS';
export const LEADERBOARD_RANGE_ADD_FAILED =
  'security/LEADERBOARD_RANGE_ADD_FAILED';

export const LEADERBOARD_RANGE_DELETE = 'security/LEADERBOARD_RANGE_DELETE';
export const LEADERBOARD_RANGE_DELETE_SUCCESS =
  'security/LEADERBOARD_RANGE_DELETE_SUCCESS';
