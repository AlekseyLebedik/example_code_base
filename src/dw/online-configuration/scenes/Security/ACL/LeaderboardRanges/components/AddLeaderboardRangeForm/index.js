import AddLeaderboardRangeForm from './container';

export default AddLeaderboardRangeForm;
