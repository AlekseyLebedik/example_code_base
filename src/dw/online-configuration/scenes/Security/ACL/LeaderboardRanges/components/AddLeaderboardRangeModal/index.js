import AddLeaderboardRangeModal from './container';

export default AddLeaderboardRangeModal;
