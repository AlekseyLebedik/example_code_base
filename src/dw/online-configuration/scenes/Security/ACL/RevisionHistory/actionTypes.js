export const REVISION_HISTORY_FETCH = 'security/REVISION_HISTORY_FETCH';
export const REVISION_HISTORY_FETCH_SUCCESS =
  'security/REVISION_HISTORY_FETCH_SUCCESS';

export const REVISION_HISTORY_REVERT = 'security/REVISION_HISTORY_REVERT';
export const REVISION_HISTORY_REVERT_SUCCESS =
  'security/REVISION_HISTORY_REVERT_SUCCESS';
