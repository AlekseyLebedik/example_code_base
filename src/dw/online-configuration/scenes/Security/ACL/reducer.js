import { combineReducers } from 'redux';
import leaderboardRangesReducer from './LeaderboardRanges/reducer';
import taskRulesReducer from './TaskRules/reducer';
import storageFilenamesReducer from './StorageFilenames/reducer';
import revisionHistoryReducer from './RevisionHistory/reducer';

export default combineReducers({
  LeaderboardRanges: leaderboardRangesReducer,
  TaskRules: taskRulesReducer,
  StorageFilenames: storageFilenamesReducer,
  RevisionHistory: revisionHistoryReducer,
});
