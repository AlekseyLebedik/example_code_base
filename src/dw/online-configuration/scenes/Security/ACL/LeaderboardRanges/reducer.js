import * as AT from './actionTypes';

const INITIAL_STATE = {
  leaderboardRanges: [],
};

const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.LEADERBOARD_RANGES_FETCH_SUCCESS:
      return {
        ...state,
        leaderboardRanges: action.leaderboardRanges,
      };
    case AT.LEADERBOARD_RANGE_DELETE_SUCCESS:
      return {
        ...state,
        leaderboardRanges: state.leaderboardRanges.filter(
          item => !action.rangeIds.includes(item.id)
        ),
      };
    default:
      return state;
  }
};

export default reducer;
