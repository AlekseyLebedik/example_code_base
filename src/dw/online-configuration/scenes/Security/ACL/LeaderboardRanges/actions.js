import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import { CriticalErrorActions } from 'dw/core/components/CriticalError';

import { actions as GlobalProgressActions } from '@demonware/devzone-core/components/GlobalProgress';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import ModalHandlers from 'dw/core/components/ModalHandlers';
import { setSelectedRowKeys } from 'dw/core/components/TableHydrated';
import * as AT from './actionTypes';

export function fetchLeaderboardRanges(params) {
  return {
    type: AT.LEADERBOARD_RANGES_FETCH,
    params,
  };
}

export function fetchLeaderboardRangesSuccess(payload) {
  return {
    type: AT.LEADERBOARD_RANGES_FETCH_SUCCESS,
    leaderboardRanges: payload.data,
  };
}

export function fetchLeaderboardRangesFailed(err, action) {
  return dispatch => {
    dispatch(GlobalProgressActions.done());
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export function addLeaderboardRange(values) {
  return dispatch => {
    dispatch({
      type: AT.LEADERBOARD_RANGE_ADD,
      values,
    });
    dispatch(ModalHandlers.submit());
  };
}

export function addLeaderboardRangeSuccess() {
  return dispatch => {
    dispatch(ModalHandlers.close());
    dispatch(
      GlobalSnackBarActions.show(
        'Leaderboard Range was successfully added.',
        'success'
      )
    );
    dispatch(fetchLeaderboardRanges());
  };
}

export function addLeaderboardRangeFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
    dispatch({
      type: AT.LEADERBOARD_RANGE_ADD_FAILED,
    });
    dispatch(ModalHandlers.stopSubmitting());
  };
}

export function deleteLeaderboardRange(rangeIds) {
  return dispatch => {
    dispatch({
      type: AT.LEADERBOARD_RANGE_DELETE,
      rangeIds,
    });
    dispatch(
      GlobalSnackBarActions.show(
        'Leaderboard Ranges deleted properly.',
        'success'
      )
    );
  };
}

export const deleteLeaderboardRangeSuccess = rangeIds => dispatch => {
  dispatch({
    type: AT.LEADERBOARD_RANGE_DELETE_SUCCESS,
    rangeIds,
  });
  dispatch(setSelectedRowKeys([]));
};

export function deleteLeaderboardRangeFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
  };
}
