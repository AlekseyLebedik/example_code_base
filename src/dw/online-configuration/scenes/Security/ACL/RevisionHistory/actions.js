import { CriticalErrorActions } from 'dw/core/components/CriticalError';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';

import { actions as GlobalProgressActions } from '@demonware/devzone-core/components/GlobalProgress';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import * as AT from './actionTypes';

export function fetchRevisions() {
  return {
    type: AT.REVISION_HISTORY_FETCH,
  };
}

export function fetchRevisionsSuccess(payload) {
  return {
    type: AT.REVISION_HISTORY_FETCH_SUCCESS,
    revisions: payload.data,
  };
}

export function fetchRevisionsFailed(err, action) {
  return dispatch => {
    dispatch(GlobalProgressActions.done());
    dispatch(CriticalErrorActions.show(err, () => action));
  };
}

export function revertRevision(revisionId) {
  return {
    type: AT.REVISION_HISTORY_REVERT,
    revisionId,
  };
}

export function revertRevisionSuccess(revisionId) {
  return dispatch => {
    dispatch({
      type: AT.REVISION_HISTORY_REVERT_SUCCESS,
      revisionId,
    });
    dispatch(
      GlobalSnackBarActions.show('Change reverted successfully.', 'success')
    );
  };
}

export function revertRevisionFailed(err) {
  return dispatch => {
    dispatch(nonCriticalHTTPError(err));
  };
}
