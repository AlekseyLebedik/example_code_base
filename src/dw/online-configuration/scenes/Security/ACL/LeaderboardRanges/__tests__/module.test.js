import { CriticalErrorActions } from 'dw/core/components/CriticalError';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import * as GlobalSnackBarActions from '@demonware/devzone-core/components/GlobalSnackBar/actions';
import ModalHandlers from 'dw/core/components/ModalHandlers';
import * as actions from '../actions';
import * as AT from '../actionTypes';
import reducer from '../reducer';

jest.mock('@demonware/devzone-core/components/GlobalSnackBar/actions');
jest.mock('dw/core/components/ModalHandlers');
jest.mock('dw/core/components/CriticalError');
jest.mock('@demonware/devzone-core/helpers/errors');

const clientType = '1';
const minId = 1;
const maxId = 2;
const id = 'Y2xpZW50X3R5cGU9MTttYXhfaWQ9MjttaW5faWQ9MQ==';
const rangeIds = [id];
const data = [
  {
    minId,
    maxId,
    clientType,
    id,
  },
  {
    minId: 2,
    maxId: 3,
    clientType,
    id: 'Y2xpZW50X3R5cGU9MTttYXhfaWQ9MzttaW5faWQ9Mg==',
  },
];
const values = {
  clientType,
  minId,
  maxId,
};
const err = 'error';

describe('Security - ACL - LeaderboardRanges', () => {
  describe('Action Creators', () => {
    const MOCKED_CRITICAL_ERROR = 'MOCKED_CRITICAL_ERROR';
    const MOCKED_NON_CRITICAL_ERROR = 'MOCKED_NON_CRITICAL_ERROR';
    const MOCKED_GLOBAL_SNACK_BAR = 'MOCKED_GLOBAL_SNACK_BAR_SHOW';
    const MOCKED_MODAL_HANDLERS_CLOSE = 'MOCKED_MODAL_HANDLERS_CLOSE';
    const MOCKED_MODAL_HANDLERS_SUBMIT = 'MOCKED_MODAL_HANDLERS_SUBMIT';
    const MOCKED_MODAL_HANDLERS_STOP_SUBMITTING =
      'MOCKED_MODAL_HANDLERS_STOP_SUBMITTING';

    beforeAll(() => {
      CriticalErrorActions.show.mockReturnValue({
        type: MOCKED_CRITICAL_ERROR,
      });
      GlobalSnackBarActions.show.mockReturnValue({
        type: MOCKED_GLOBAL_SNACK_BAR,
      });
      ModalHandlers.close.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_CLOSE,
      });
      ModalHandlers.submit.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_SUBMIT,
      });
      ModalHandlers.stopSubmitting.mockReturnValue({
        type: MOCKED_MODAL_HANDLERS_STOP_SUBMITTING,
      });
      nonCriticalHTTPError.mockReturnValue({
        type: MOCKED_NON_CRITICAL_ERROR,
      });
    });

    it('LEADERBOARD_RANGES_FETCH', () => {
      expect(actions.fetchLeaderboardRanges()).toHaveProperty(
        'type',
        AT.LEADERBOARD_RANGES_FETCH
      );
    });

    it('LEADERBOARD_RANGES_FETCH_SUCCESS', () => {
      const action = actions.fetchLeaderboardRangesSuccess({ data });

      expect(action).toHaveProperty(
        'type',
        AT.LEADERBOARD_RANGES_FETCH_SUCCESS
      );
      expect(action).toHaveProperty('leaderboardRanges', data);
    });

    it('fetchLeaderboardRangesFailed', () => {
      const action = actions.fetchLeaderboardRangesFailed();

      expect(action).toAsyncDispatch({ type: MOCKED_CRITICAL_ERROR });
    });

    it('LEADERBOARD_RANGE_ADD', () => {
      const action = actions.addLeaderboardRange(values);

      expect(action).toAsyncDispatch({
        type: AT.LEADERBOARD_RANGE_ADD,
        values,
      });
      expect(action).toAsyncDispatch({ type: MOCKED_MODAL_HANDLERS_SUBMIT });
    });

    it('addLeaderboardRangeSuccess', () => {
      const action = actions.addLeaderboardRangeSuccess();

      expect(action).toAsyncDispatch({ type: MOCKED_MODAL_HANDLERS_CLOSE });
      expect(action).toAsyncDispatch({ type: MOCKED_GLOBAL_SNACK_BAR });
      expect(action).toAsyncDispatch({ type: AT.LEADERBOARD_RANGES_FETCH });
    });

    it('addLeaderboardRangeFailed', () => {
      const action = actions.addLeaderboardRangeFailed(err);

      expect(action).toAsyncDispatch({ type: MOCKED_NON_CRITICAL_ERROR });
      expect(action).toAsyncDispatch({ type: AT.LEADERBOARD_RANGE_ADD_FAILED });
      expect(action).toAsyncDispatch({
        type: MOCKED_MODAL_HANDLERS_STOP_SUBMITTING,
      });
    });

    it('LEADERBOARD_RANGE_DELETE', () => {
      const action = actions.deleteLeaderboardRange(rangeIds);

      expect(action).toAsyncDispatch({
        type: AT.LEADERBOARD_RANGE_DELETE,
        rangeIds,
      });
      expect(action).toAsyncDispatch({ type: MOCKED_GLOBAL_SNACK_BAR });
    });

    it('deleteLeaderboardRangeSuccess', () => {
      const action = actions.deleteLeaderboardRangeSuccess(rangeIds);

      expect(action).toAsyncDispatch({
        type: AT.LEADERBOARD_RANGE_DELETE_SUCCESS,
        rangeIds,
      });
    });

    it('deleteLeaderboardRangeFailed', () => {
      const action = actions.deleteLeaderboardRangeFailed(err);

      expect(action).toAsyncDispatch({ type: MOCKED_NON_CRITICAL_ERROR });
    });
  });

  describe('Reducer', () => {
    it('return the initial state', () => {
      expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('handle LEADERBOARD_RANGES_FETCH_SUCCESS', () => {
      expect(
        reducer(undefined, actions.fetchLeaderboardRangesSuccess({ data }))
      ).toMatchSnapshot();
    });

    it('handle LEADERBOARD_RANGE_DELETE_SUCCESS', () => {
      const state = reducer(
        undefined,
        actions.fetchLeaderboardRangesSuccess({ data })
      );
      expect(
        reducer(state, { type: AT.LEADERBOARD_RANGE_DELETE_SUCCESS, rangeIds })
      ).toMatchSnapshot();
    });
  });
});
