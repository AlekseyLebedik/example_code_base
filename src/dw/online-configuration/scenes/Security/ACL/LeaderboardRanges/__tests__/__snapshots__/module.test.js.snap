// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security - ACL - LeaderboardRanges Reducer handle LEADERBOARD_RANGE_DELETE_SUCCESS 1`] = `
Object {
  "leaderboardRanges": Array [
    Object {
      "clientType": "1",
      "id": "Y2xpZW50X3R5cGU9MTttYXhfaWQ9MzttaW5faWQ9Mg==",
      "maxId": 3,
      "minId": 2,
    },
  ],
}
`;

exports[`Security - ACL - LeaderboardRanges Reducer handle LEADERBOARD_RANGES_FETCH_SUCCESS 1`] = `
Object {
  "leaderboardRanges": Array [
    Object {
      "clientType": "1",
      "id": "Y2xpZW50X3R5cGU9MTttYXhfaWQ9MjttaW5faWQ9MQ==",
      "maxId": 2,
      "minId": 1,
    },
    Object {
      "clientType": "1",
      "id": "Y2xpZW50X3R5cGU9MTttYXhfaWQ9MzttaW5faWQ9Mg==",
      "maxId": 3,
      "minId": 2,
    },
  ],
}
`;

exports[`Security - ACL - LeaderboardRanges Reducer return the initial state 1`] = `
Object {
  "leaderboardRanges": Array [],
}
`;
